demos/classification_benchmark_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,134 @@
+            out_scored_labels_to_append.emplace_back(conf_index_it->first,
+                                                     *conf_index_it->second,
+                                                     recalculated_label_id != 0 ? in_labes.at(recalculated_label_id) : "others");
+        } catch (const std::out_of_range& ex) {
+            throw std::out_of_range(std::string("Provided labels file doesn't contain classified label index\nException: ") + ex.what());
+        }
+    }
+}
+
+IndexScore IndexScore::create_from_array(const float *out_blob_data_ptr, size_t out_blob_element_count,
+                                         size_t top_k_amount) {
+    IndexScore ret;
+    if (!out_blob_data_ptr || !out_blob_element_count || !top_k_amount) {
+        return IndexScore();
+    }
+    // find top K
+    size_t i = 0;
+    // fill & sort topK with first K elements of N array: O(K*Log(K))
+    while(i < std::min(top_k_amount, out_blob_element_count)) {
+        ret.max_element_indices.push_back(i);
+        ret.max_confidence_with_indices.emplace(out_blob_data_ptr[i], std::prev(ret.max_element_indices.end()));
+        i++;
+    }
+
+    // continue searching K elements through remnant N-K array elements
+    // greater than the minimum element in the pivot topK
+    // O((N-K)*Log(K))
+    for (; i < out_blob_element_count; i++) {
+        const auto &low_confidence_it = ret.max_confidence_with_indices.begin();
+        if (out_blob_data_ptr[i] >= low_confidence_it->first) {
+            auto list_min_elem_it = low_confidence_it->second;
+            *list_min_elem_it = i;
+            ret.max_confidence_with_indices.erase(low_confidence_it);
+            ret.max_confidence_with_indices.emplace(out_blob_data_ptr[i], list_min_elem_it);
+        }
+    }
+
+    ret.classification_indices_scale = out_blob_element_count;
+    return ret;
+}
+
+GAPI_OCV_KERNEL(OCVLocateROI, custom::LocateROI) {
+    // This is the place where we can run extra analytics
+    // on the input image frame and select the ROI (region
+    // of interest) where we want to classify our objects (or
+    // run any other inference).
+    //
+    // Crops the input image to square (this is
+    // the most convenient aspect ratio for classificators to use)
+
+    static void run(const cv::Size& in_size,
+                    cv::Rect &out_rect) {
+
+        // Identify the central point & square size (- some padding)
+        const auto center = cv::Point{in_size.width/2, in_size.height/2};
+        auto sqside = std::min(in_size.width, in_size.height);
+
+        // Now build the central square ROI
+        out_rect = cv::Rect{ center.x - sqside/2
+                             , center.y - sqside/2
+                             , sqside
+                             , sqside
+                            };
+    }
+};
+
+GAPI_OCV_KERNEL(OCVTopK, custom::TopK) {
+    static void run(const cv::MediaFrame &in, const cv::Mat &out_blob, uint32_t top_k_amount, IndexScore &out) {
```

Comment:
Why do we need `cv::MediaFrame &in` there?

Reply:
not sure, does graph can be constructed without source frame/mat linkage?

Reply:
You have source linkage (in infer at least) but there you don't need this source at all